liblocation/src/main/java/com/mapbox/android/core/location/LocationPendingIntentProvider.java

```diff
@@ -0,0 +1,40 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+
+import java.util.concurrent.CopyOnWriteArrayList;
+
+class LocationPendingIntentProvider {
+  private Context context;
+  private SdkChecker sdkChecker;
+  private CopyOnWriteArrayList<LocationEngineListener> locationEngineListeners;
+
+  LocationPendingIntentProvider(Context context, SdkChecker sdkChecker,
+                         CopyOnWriteArrayList<LocationEngineListener> locationEngineListeners) {
+    this.context = context;
+    this.sdkChecker = sdkChecker;
+    this.locationEngineListeners = locationEngineListeners;
+  }
+
+  PendingIntent buildPendingIntent() {
```

Comment:
We can build the `PendingIntent` in the constructor so we don't have to store the `Context`, checker, etc. in this class.